### PR TITLE
Backport v3.2 from v3.3

### PR DESCRIPTION
**Pull request by emphazer**
i tried to focus on critical changes.
i skipped https://github.com/coreruleset/coreruleset/pull/1805 for example.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -5,6 +5,31 @@
   or the CRS mailinglist at
 * https://lists.owasp.org/mailman/listinfo/owasp-modsecurity-core-rule-set
 
+== Version 3.2.4 - 2023-10-27 ==
+
+Important changes:
+ * Backported important Fixes from v3.3
+
+Fixes and improvements:
```
Comment (airween) [inline CHANGES:13]: As I see you made the fixes which described "CRS bypass" issues, which is fine for me.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
-![Travis build v3.2/dev](https://img.shields.io/travis/spiderlabs/owasp-modsecurity-crs/v3.2/dev?label=v3.2%2Fdev)
-![Travis build v3.1/dev](https://img.shields.io/travis/spiderlabs/owasp-modsecurity-crs/v3.1/dev?label=v3.1%2Fdev)
-![Travis build v3.0/dev](https://img.shields.io/travis/spiderlabs/owasp-modsecurity-crs/v3.0/dev?label=v3.0%2Fdev)
-[![OWASP Flagship](https://img.shields.io/badge/owasp-flagship%20project-38a047.svg)](https://www.owasp.org/index.php/OWASP_Project_Inventory#tab=Flagship_Projects)
+![GHA build v3.3/dev](https://github.com/coreruleset/coreruleset/workflows/Test%20Ruleset%20against%20matrix/badge.svg?branch=v3.3%2Fdev)
```
Comment (airween) [inline README.md:1]: Why do we want to add the next version's URL into the README? Is there any reason?

File: rules/REQUEST-921-PROTOCOL-ATTACK.conf [hunk 3]
```diff
@@ -24,14 +24,14 @@ SecRule TX:EXECUTING_PARANOIA_LEVEL "@lt 1" "id:921012,phase:2,pass,nolog,skipAf
...
-# This rule looks for a CR/LF character in combination with a HTTP / WEBDAV method name.
+# This rule looks for a HTTP / WEBDAV method name in combination with the word http/\d or a CR/LF character.
 # This would point to an attempt to inject a 2nd request into the request, thus bypassing
 # tests carried out on the primary request.
 #
 # [ References ]
 # http://projects.webappsec.org/HTTP-Request-Smuggling
 #
-SecRule ARGS_NAMES|ARGS|XML:/* "@rx [\n\r]+(?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\s+[^\s]+(?:\s+http|[\r\n])" \
+SecRule ARGS_NAMES|ARGS|REQUEST_BODY|XML:/* "@rx (?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\s+[^\s]+(?:\s+http\/\d|[\r\n])" \
```
Comment (airween) [inline rules/REQUEST-921-PROTOCOL-ATTACK.conf:34]: I think this modification is not complete, a part of the new regex is missing after `unlock)`: `(?:\/|\w)`. The line in 3.3 is this:

```
SecRule ARGS_NAMES|ARGS|REQUEST_BODY|XML:/* "@rx (?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\s+(?:\/|\w)[^\s]*(?:\s+http\/\d|[\r\n])" \
```

File: rules/REQUEST-921-PROTOCOL-ATTACK.conf [hunk 4]
```diff
@@ -84,7 +84,7 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAME
     setvar:'tx.anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
 
 
-SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx (?:\bhttp\/(?:0\.9|1\.[01])|<(?:html|meta)\b)" \
+SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx (?:\bhttp\/\d|<(?:html|meta)\b)" \
```
Comment (airween) [inline rules/REQUEST-921-PROTOCOL-ATTACK.conf:87]: I think there is an extra `\` character here before the `/\d`. In 3.3 the operator's argument is [this](https://github.com/coreruleset/coreruleset/blob/v3.3/master/rules/REQUEST-921-PROTOCOL-ATTACK.conf#L89):

```
(?:\bhttp/\d|<(?:html|meta)\b)
```
(and not `(?:\bhttp\/\d|<(?:html|meta)\b)`)

File: rules/REQUEST-921-PROTOCOL-ATTACK.conf [hunk 5]
```diff
@@ -187,6 +187,33 @@ SecRule ARGS_GET_NAMES|ARGS_GET "@rx (?:\n|\r)+(?:\s|location|refresh|(?:set-)?c
...
+    phase:1,\
+    block,\
+    t:none,t:urlDecodeUni,\
+    msg:'HTTP Splitting (CR/LF in request filename detected)',\
+    logdata:'Matched Data: %{TX.0} found within %{MATCHED_VAR_NAME}: %{MATCHED_VAR}',\
+    tag:'application-multi',\
+    tag:'language-multi',\
+    tag:'platform-multi',\
+    tag:'attack-protocol',\
+    tag:'paranoia-level/1',\
```
Comment (airween) [inline rules/REQUEST-921-PROTOCOL-ATTACK.conf:207]: We do not have `paranoia-level/N` tags in 3.2, you should remove this tag.

File: rules/REQUEST-921-PROTOCOL-ATTACK.conf [hunk 6]
```diff
@@ -187,6 +187,33 @@ SecRule ARGS_GET_NAMES|ARGS_GET "@rx (?:\n|\r)+(?:\s|location|refresh|(?:set-)?c
...
+    tag:'OWASP_CRS',\
+    tag:'OWASP_CRS/WEB_ATTACK/HTTP_SPLITTING',\
+    ctl:auditLogParts=+E,\
+    ver:'OWASP_CRS/3.2.0',\
```
Comment (airween) [inline rules/REQUEST-921-PROTOCOL-ATTACK.conf:211]: Here you should add tag `OWASP_CRS/3.2.3`, not `3.2.0`.

File: rules/REQUEST-931-APPLICATION-ATTACK-RFI.conf [hunk 7]
```diff
@@ -121,7 +121,7 @@ SecRule ARGS "@rx ^(?i:file|ftps?|https?)://(.*)$" \
     severity:'CRITICAL',\
     setvar:'tx.rfi_parameter_%{MATCHED_VAR_NAME}=%{tx.1}',\
```
Comment (airween) [inline rules/REQUEST-931-APPLICATION-ATTACK-RFI.conf:122]: An extra `.` (dot) is missing from here. The correct action is:

```
setvar:'tx.rfi_parameter_%{MATCHED_VAR_NAME}=.%{tx.1}',\
```

**Review by airween — CHANGES_REQUESTED** [review]
I reviewed the modifications, see the detailed comments above.

**Comment by theseion** [comment]
We have decided to close this PR and focus on the v4 release.

**Pull request closed without merging**